### ISSUE TITLE
How to install community netinfo

### ISSUE THREAD

**Issue by rraczae** [microsoft/react-native-windows#7390]
The documentation says tp update `MainReactNativeHost.cs` like this:

```
+ using ReactNativeCommunity.NetInfo;
......
        protected override List<IReactPackage> Packages => new List<IReactPackage>
        {
            new MainReactPackage(),
+           new RNCNetInfoPackage(),
        };
```

However that file does not exist. Any hints?

Thanks!

**Comment by chrisglein** [comment]
@rraczae Can you link to the page in the documentation you were referring to?

**Comment by asklar** [comment]
@chrisglein I believe they are referring to this: https://github.com/react-native-netinfo/react-native-netinfo#link-c-implementation-1

@rraczae what version of react-native and react-native-windows are you using?

**Comment by asklar** [comment]
Looked at this some more. The NetInfo module has 2 Windows implementations: RNCInfo and RNCInfoCpp.
The RNCInfo implementation is for the C# React Native Windows which has been unsupported for over a year (and will not work with current RNW versions). The RNCInfoCpp project is the one you should be using.

**Comment by rraczae** [comment]
Sorry guys, I was not getting notifications for this... Thanks that explains! Where does it say c# is unsupported? I was still able to pass it as a language option.

**Comment by asklar** [comment]
@rraczae yeah that's a confusing world we live in :) Let me elaborate:
Once upon a time, we had a RNW framework written in C# - this one is deprecated and modules written for this version are deprecated too. We deleted the code back in 0.60 I think?

We rewrote the RNW framework in C++, and you can write modules and apps against this version of RNW in either C++ or C# (this is the option you see in the CLI to create an app/module).
The part that is deprecated/deleted is the "RNW is written in C#" and modules that depend on the APIs that existed in that version. 
Hope this clarifies!

**Comment by chrisglein** [comment]
@rraczae Did that help clarify? Are you unblocked now?

**Comment by rraczae** [comment]
@chrisglein, @asklar thank you for the explanation, yes that definitely helps! 

As far as unblocked, unfortunately I don't see a way right now how we could add windows as a third target to our recently built app. Long story short, we started with expo, we ejected and we have an android/ios app targeting tablets. Everything works great. The request now is to explore surface/win10 support hence we're looking at React Native Windows. Unfortunately we seem to have a bunch of compatibility issues, NetInfo was just the first one. With cpp it was pretty easy to get it working but the everything just snowballed.

Bottom line, even if we could get it to work with all the modules we use, it seems like adding windows to the project would make everything a lot more fragile. At the moment we are leaning towards rebuilding the "desktop" version in electron and rather maintain two separate code bases. 

Truly appreciate the quick responses here and would really like to hear your take on our approach as well. Also the windows version is still in early planning stage, realistically we would start developing a few months from now so we can definitely revisit.

Best!

Robert

**Comment by chrisglein** [comment]
> Unfortunately we seem to have a bunch of compatibility issues, NetInfo was just the first one. With cpp it was pretty easy to get it working but the everything just snowballed.

It'd be super helpful to us to know which other compatibility issues you've hit. We've been trying to gather those together and drive priority on what is blocking people. Are these other modules like NetInfo? Core components that aren't working? Non-RN Windows stuff?

We haven't been able to close the gap on Expo support yet, which sounds like is what would really help you. But even knowing that is helpful for our prioritization. We've grown the set of community modules that support Windows quite a bit over the past year but haven't yet done any Expo modules.

**Comment by rraczae** [comment]
@chrisglein sure! I will have some more time over the weekend and I can gather more info than just sending package.json. 

About Expo, I think supporting most expo libraries would be a huge step forward. It is a really common use case that people start in expo so they can make quick progress and then eject.

About c#, that by itself can be a stopper for many. I am ok working in c++ although I have not been using it regularly for while now. But most of the people in mobile/web world are way more comfortable and productive in c#. So NetInfo not being ready to be used in a c# project is a stopper at my company outside of my control.

**Comment by asklar** [comment]
@rraczae Just to clarify: you *can* still use the netinfo module (the C++ version) in a C# app written in RNW. The thing that's deprecated for C# is an old version of RNW where the framework was written in C# itself. 
The current version of RNW (written in C++) supports modules and apps written in either C# or C++.

**Comment by rraczae** [comment]
Cool, then I just have to look into this more. I can add the project and the reference to the lib, but how would I register the cpp package in the C# app?

**Comment by asklar** [comment]
Yes, if you add the module to your package.json, run yarn, then autolinking will take care of the rest (it will add the right project references and generate code to register the package for you). You can run autolinking from the command line `npx react-native autolink-windows`, or when you build with `npx react-native run-windows` it will run automatically for you. Autolinking doesn't run if you are building inside Visual Studio.

**Comment by rraczae** [comment]
Hmm. I did all that and ran autolinking multiple times yet it failed to resolve... the missing module exception only disappeared when i went through the manual steps (adding a reference to the project under node modules to the solution, manually registering the package)

**Comment by asklar** [comment]
Which version of RNW are you using? We just released 0.64, can you try that? It should work out of the box, if it does not that's something we need to fix : )

**Comment by rraczae** [comment]
Yessir I'll give that a shot over the weekend! I used .63. Thanks for the guidance much appreciated! Here are all my packages for the reference:

```
  "dependencies": {
    "@expo/vector-icons": "^12.0.2",
    "@react-native-community/datetimepicker": "3.0.4",
    "@react-native-community/hooks": "^2.6.0",
    "@react-native-community/masked-view": "0.1.10",
    "@react-native-community/netinfo": "5.9.10",
    "@react-native-firebase/analytics": "^10.8.1",
    "@react-native-firebase/app": "^10.8.1",
    "@react-native-firebase/crashlytics": "^10.8.1",
    "@react-navigation/bottom-tabs": "^5.11.2",
    "@react-navigation/native": "^5.8.10",
    "@react-navigation/stack": "^5.12.8",
    "@types/react-redux": "^7.1.15",
    "amazon-cognito-identity-js": "^4.5.10",
    "aws-amplify": "^3.3.14",
    "aws-amplify-react-native": "^4.3.0",
    "eslint-config-react": "^1.1.7",
    "eslint-plugin-jest": "^24.1.3",
    "expo": "~40.0.0",
    "expo-asset": "~8.2.1",
    "expo-constants": "~9.3.5",
    "expo-font": "~8.4.0",
    "expo-linear-gradient": "~8.4.0",
    "expo-linking": "^2.0.0",
    "expo-print": "~9.2.0",
    "expo-sharing": "~8.5.0",
    "expo-splash-screen": "~0.8.1",
    "expo-status-bar": "~1.0.3",
    "expo-updates": "~0.4.0",
    "expo-web-browser": "~8.6.0",
    "fast-deep-equal": "^3.1.3",
    "metro-react-native-babel-preset": "^0.65.2",
    "react": "16.13.1",
    "react-dom": "17.0.1",
    "react-hook-form": "^6.13.1",
    "react-native": "~0.63.4",
    "react-native-gesture-handler": "~1.9.0",
    "react-native-image-pan-zoom": "^2.1.12",
    "react-native-modal": "^11.6.1",
    "react-native-reanimated": "~1.13.0",
    "react-native-safe-area-context": "3.1.9",
    "react-native-screens": "~2.16.1",
    "react-native-svg": "12.1.0",
    "react-native-unimodules": "~0.12.0",
    "react-native-web": "~0.14.9",
    "react-native-windows": "^0.63.0-0",
    "react-redux": "^7.2.2",
    "redux": "^4.0.5"
  },
  "devDependencies": {
    "@babel/core": "~7.12.10",
    "@types/base64-js": "^1.3.0",
    "@types/react": "^17.0.0",
    "@types/react-native": "~0.63.40",
    "@typescript-eslint/eslint-plugin": "^4.10.0",
    "@typescript-eslint/parser": "^4.10.0",
    "babel-plugin-module-resolver": "^4.1.0",
    "eslint": "^7.15.0",
    "eslint-config-prettier": "^7.0.0",
    "eslint-config-universe": "^6.0.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.3.0",
    "eslint-plugin-react": "^7.21.5",
    "eslint-plugin-react-hooks": "^4.2.0",
    "eslint-plugin-react-native": "^3.10.0",
    "husky": "^5.0.6",
    "jest-expo": "~40.0.1",
    "prettier": "^2.2.1",
    "typescript": "~4.1.3",
    "babel-jest": "~25.2.6",
    "jest": "~25.2.6",
    "react-test-renderer": "~16.13.1"
  },
```
